### DISCUSSION TITLE
Get all source files from program

### DISCUSSION THREAD

**Discussion by lastmjs** [swc-project/swc#5531] [Q&A] [answered]
Hi, we're rewriting a TypeScript compiler project in Rust and using SWC for the TypeScript AST. Using `tsc`, once I have a program I can easily get all source files referenced from the imports of the entrypoint file like this:

```typescript
const program = tsc.createProgram(['index.ts'], {});

const sourceFiles = program.getSourceFiles();
```
How can I easily achieve the same thing using `swc_ecma_parser`?

Here's the code I have so far:

```rust
let filepath = Path::new(&ts_file_path_string_value).to_path_buf();

let cm: Lrc<SourceMap> = Default::default();

let fm = cm.load_file(&filepath).unwrap();

let lexer = Lexer::new(
    Syntax::Typescript(TsConfig::default()),
    Default::default(),
    StringInput::from(&*fm),
    None
);

let mut parser = Parser::new_from(lexer);

let program = parser.parse_program().unwrap();
```

The `program` I have is just for one source file. Is there a way to get all source files like I can do in `tsc`?

**Comment by kdy1** [comment, marked as answer]
You need to resolve dependencies by yourself. swc works file-by-file
